scheduler.go

```diff
@@ -298,7 +298,15 @@ func (s *scheduler) selectExecJobsOutForRescheduling(id uuid.UUID) {
 		// so we don't need to reschedule it.
 		return
 	}
-	j.lastScheduledRun = j.nextScheduled
+	if len(j.nextScheduled) > 0 {
+		// always grab the last element in the slice as that is the furthest
+		// out in the future and the time from which we want to calculate
+		// the subsequent next run time.
+		slices.SortStableFunc(j.nextScheduled, func(a, b time.Time) int {
+			return a.Compare(b)
+		})
+		j.lastScheduledRun = j.nextScheduled[len(j.nextScheduled)-1]
```

Comment:
Is this being sorted by increasing time? Do you want the first element in this list?

Reply:
it's sorted increasing, so `now, now+1hr, now+2hr`. We're scheduling the next run here, so we want the greatest time in the slice, the last value.